src/app/libraries/csrf_lib.php

```diff
@@ -20,12 +20,20 @@ public static function generate_token() {
      * @return string The token
      */
     public function new_csrf_token() {
-        $csrf_token = self::generate_token();
-        $_SESSION['csrf_token'] = $csrf_token;
-        return $csrf_token;
+        if ( ! isset($_SESSION['csrf_token'])) {
```

Comment:
Space before the `!` too? That sorta goes against convention. But meh, whatever. :package: